backend/app/tests/api/routes/theme/test_post_theme_api.py

```diff
@@ -0,0 +1,92 @@
+from fastapi.testclient import TestClient
+
+from app.core.config import settings
+from app.models.theme import MAX_NAME_LENGTH, MIN_NAME_LENGTH
+
+
+def test_post_request_to_theme_route_should_create_theme_when_theme_does_not_exist(
+    client: TestClient, superuser_token_headers: dict[str, str]
+) -> None:
+    theme_name = "Leadership"
+    description = "Focus on qualities of leadership, such as guidance, influence, vision, and the ability to inspire and empower others."
+
+    data = {"name": theme_name, "description": description}
+    response = client.post(
+        f"{settings.API_V1_STR}/themes/",
+        headers=superuser_token_headers,
+        json=data,
+    )
+    assert response.status_code == 200
+    content = response.json()
+    assert content["name"] == data["name"]
+    assert content["description"] == data["description"]
+    assert "id" in content
+
+
+def test_post_request_to_theme_route_should_return_4xx_when_theme_already_exists(
+    client: TestClient, superuser_token_headers: dict[str, str]
+) -> None:
+    theme_name = "Management"
+    description = "Organizing, planning, and overseeing tasks, projects, or teams."
+
+    data = {"name": theme_name, "description": description}
+    response = client.post(
+        f"{settings.API_V1_STR}/themes/",
+        headers=superuser_token_headers,
+        json=data,
+    )
+    assert response.status_code == 200
```

Comment:
note how you have sort of duplicated code in these UTs around creating the data packet, making the request, then asserting on response code. You may consider refactoring those operations into a helper.

Reply:
I've created a small helper function and also consolidated the three separate tests into one using @pytest.mark.parametrize